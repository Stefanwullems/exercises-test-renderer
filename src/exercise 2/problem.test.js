import React, { useState } from 'react'
import { create, act } from 'react-test-renderer'

const Button = (props) => {
  const [count, setCount] = useState(0)

  return <button onClick={() => setCount(count + 1)}>{count}</button>
}

// Make the tests pass by only using toJson and accessing props

test('The counter updates when I click the button', () => {
  const { toJSON } = create(<Button key={1} />)
  let button
  /* code here */

  // initialize button
  button = undefined

  act(() => {
    // trigger onClick

  })

  // re-render
  button = undefined

  /* end */

  expect(button.children[0]).toBe('1')
  expect(button).toMatchSnapshot()
})
