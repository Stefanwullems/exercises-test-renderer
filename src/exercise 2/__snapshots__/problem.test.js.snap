// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The counter updates when I click the button 1`] = `
<button
  onClick={[Function]}
>
  1
</button>
`;
